examples/timestamping/backend/src/lib.rs

```diff
@@ -39,35 +39,45 @@ pub use crate::{
 };
 
 use exonum::{
-    merkledb::BinaryValue,
+    merkledb::{validation::is_valid_index_name_component, BinaryValue},
     runtime::{CommonError, ExecutionContext, ExecutionError},
 };
 use exonum_derive::{ServiceDispatcher, ServiceFactory};
 use exonum_rust_runtime::{api::ServiceApiBuilder, Service};
+use exonum_supervisor::Configure;
 
 use crate::{api::PublicApi as TimestampingApi, schema::Schema};
 
 #[derive(Debug, ServiceDispatcher, ServiceFactory)]
-#[service_dispatcher(implements("TimestampingInterface"))]
+#[service_dispatcher(implements("TimestampingInterface", raw = "Configure<Params = Config>"))]
 #[service_factory(proto_sources = "proto")]
 pub struct TimestampingService;
 
+fn verify_config(context: &ExecutionContext<'_>, config: &Config) -> Result<(), ExecutionError> {
+    if !is_valid_index_name_component(&config.time_service_name) {
+        return Err(Error::InvalidConfig.into());
+    }
+
+    if context
```

Comment:
I'd argue that this check is still not strong enough given that the `timestamp` method implementation uses `BlockchainData::service_schema`, which gives significantly higher guarantees:

- The service instance exists and is active
- The instance corresponds to a supported version of the time service artifact

These checks could be performed either by instantiating the time service schema as it's done in the `timestamp` handler, or by using `impl RequireArtifact for TimeSchema` directly.

There are marginal cases where the check may be insufficient (e.g., the time service is switched off in the same proposal) or too strict (the time service is switched on in the same proposal), but I'm not sure whether we should consider them.